lib/irb/cmd/debug.rb

```diff
@@ -82,14 +82,22 @@ def DEBUGGER__.capture_frames(*args)
       # it's a bundled gem. This method tries to activate and load that.
       def load_bundled_debug_gem
         # Discover latest debug.gem under GEM_PATH
-        debug_gem = Gem.paths.path.map { |path| Dir.glob("#{path}/gems/debug-*") }.flatten.select do |path|
-          File.basename(path).match?(/\Adebug-\d+\.\d+\.\d+\z/)
+        debug_gem = Gem.paths.path.flat_map { |path| Dir.glob("#{path}/gems/debug-*") }.select do |path|
```

Comment:
Nitpick: It looks like `Gem.paths.path` can be stored in a local?

Reply:
What do you mean? Are you talking about assigning it to a local variable? `Gem.paths` is memoized and `.path` is just an attr_reader. Do you still want to do that?

Reply:
Ah it's memoized. There's no need then 👍